tests/feature_resolution_editions.js

```diff
@@ -0,0 +1,524 @@
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'NONE',
+        '(abc)': { d_e: 'deeply_nested_false' } 
+    })
+
+    test.end();
+});
+
+tape.test("feature resolution inheritance message to extensions", function(test) {
+
+    var rootEditionsOverriden = protobuf.parse(`
+    edition = "2023";
+    option features.json_format = LEGACY_BEST_EFFORT;
+    option features.(abc).d_e = deeply_nested_false;
+
+    message Message {
+        option features.utf8_validation = NONE;
+        extend Message {
+            int32 bar = 10 [features.utf8_validation = VERIFY];
+        }
+    }`).root.resolveAll();
+
+    test.same(rootEditionsOverriden.lookup(".bar")._features, {
+        enum_type: 'OPEN',
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_false' } 
+    })
+
+    test.end();
+});
+
+tape.test("feature resolution inheritance message to enum", function(test) {
+
+    var rootEditionsOverriden = protobuf.parse(`edition = "2023";
+    option features.json_format = LEGACY_BEST_EFFORT;
+    option features.(abc).d_e = deeply_nested_false;
+    message Message {
+        option features.utf8_validation = NONE;
+        enum SomeEnum {
+            ONE = 1;
+            TWO = 2;
+        }
+    }`).root.resolveAll();
+
+    test.same(rootEditionsOverriden.lookup("Message").lookup("SomeEnum")._features, {
+        enum_type: 'OPEN',
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'NONE',
+        '(abc)': { d_e: 'deeply_nested_false' } 
+    })
+
+    test.end();
+});
+
+tape.test("feature resolution inheritance file to enum", function(test) {
+
+    var rootEditionsOverriden = protobuf.parse(`edition = "2023";
+    option features.json_format = LEGACY_BEST_EFFORT;
+    option features.(abc).d_e = deeply_nested_false;
+    enum SomeEnum {
+        option features.utf8_validation = NONE;
+        ONE = 1;
+        TWO = 2;
+    }`).root.resolveAll();
+
+    test.same(rootEditionsOverriden.lookup("SomeEnum")._features, {
+        enum_type: 'OPEN',
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'NONE',
+        '(abc)': { d_e: 'deeply_nested_false' } 
+    })
+
+    test.end();
+});
+
+tape.test("feature resolution inheritance file to service and service to method", function(test) {
+    var rootEditionsOverriden = protobuf.parse(`edition = "2023";
+    option features.json_format = LEGACY_BEST_EFFORT;
+    option features.(abc).d_e = deeply_nested_false;
+    service MyService {
+        option features.json_format = ALLOW;
```

Comment:
This is also the 2023 default, so this test isn't quite locking down all the behaviors